pkg/metrics3/client.go

```diff
@@ -0,0 +1,464 @@
+package metrics3
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	AccessKey           string = "api"
+	DataCollectionToken string = "data"
+)
+
+type AnodotToken struct {
+	Value string
+	Type  string
+}
+
+func NewAnoToken(token string, ttype string) (*AnodotToken, error) {
+	if len(strings.TrimSpace(token)) == 0 {
+		return nil, fmt.Errorf("token can't be blank")
+	}
+
+	switch ttype {
+	case AccessKey, DataCollectionToken:
+		return &AnodotToken{Value: token, Type: ttype}, nil
+	}
+
+	return nil, fmt.Errorf("token type can be DataCollectionToken or AccessKey")
+}
+
+type AnodotResponse interface {
+	HasErrors() bool
+	ErrorMessage() string
+	RawResponse() *http.Response
+}
+
+type Api30Response struct {
+	Error *struct {
+		Status        int    `json:"status"`
+		Name          string `json:"name"`
+		Message       string `json:"message"`
+		AndtErrorCode int    `json:"andtErrorCode"`
+		Path          string `json:"path"`
+	}
+	HttpResponse *http.Response `json:"-"`
+}
+
+func (r *Api30Response) HasErrors() bool {
+	return r.Error != nil
+}
+
+func (r *Api30Response) ErrorMessage() string {
+	return fmt.Sprintf("%+v\n", r.Error)
+}
+
+func (r *Api30Response) RawResponse() *http.Response {
+	return r.HttpResponse
+}
+
+type RefreshBearerResponse struct {
+	Bearer string
```

Comment:
i think it might be better to add refresh time here as a filed

Reply:
That's should be an internal struct. And the user won't see it, because token refreshment is being processed on the lib side.